### PR TITLE
Updated asset/pages and scss files

### PR DESCRIPTION
**Pull request by kenjdavidson**
Resolution for issue #37 
- Moved `404.html`, `allpages.html`, `search.json` into the `/assets` folder and confirmed that they are now being exported to the _site build.
- Split the `main.scss` up where `main.scss` contains only the imports, while `_sass/_jekyll-doc-theme.scss` contain all the them specific styles.

### CODE REVIEW

**Comment by kenjdavidson** [comment]
Also updated the footer style so that when the footer becomes two lines, it's displayed properly where the background reaches the bottom of the window.

**Comment by aksakalli** [comment]
Thank you. I think it is not so intuitive to see page sources inside `assets`. Alternatively, we can put the rendering logic to `_layouts` and have an empty page in the root? Like this theme:

* https://github.com/mmistakes/so-simple-theme/blob/master/_layouts/tag.html
* https://github.com/mmistakes/so-simple-theme/blob/master/example/tags.md

Also, it would be nice to include the documentation, how to install it as an remote theme. What do you think?

**Comment by kenjdavidson** [comment]
It's your show, but in all honesty after going through the `Remote-Theme` and the `Jekyll` code, it looks like this was the design when they moved to the `/asset` directory.  The function called is actually `LoadThemeAssets` and not just `LoadAssets`.  

If you relate this specifically to this project, you would have:

- ./_layout/allposts.html
- ./allposts.md

this will still not build the `allposts.html` file from the theme, the user of the theme will still need to copy the `./allposts.md` file from the project, to their own project.   I think as long as it's documented that the assets folder contains the three pages (and maybe more in the future) that the person implementing the theme doesn't need to do anything.

In the project I'm using your theme it's working perfectly.  I don't need to think about it, and if you updated the `./assets/allposts.md` with anything, I just need to rebuild, I don't need to recopy `./allposts.md` to my project again.   I also have the ability to override my own `./asset/allposts.md` if I do want it to change.

All in all, obviously it's your show (and I'm just one guy), but after going through the code and seeing what they had actually called the assets folder, it makes sense.

File: assets/css/main.scss [hunk 1]
```diff
@@ -19,68 +17,5 @@
   {% endif %}
 
   "syntax-highlighting",
-  "typeahead"
```
Comment (aksakalli) [inline assets/css/main.scss:22]: why did you remove this one?

Comment (kenjdavidson) [reply assets/css/main.scss:22]: That would be an accident, I meant to move it into the `_scss/jekyll-doc-theme.scss` file to keep the main.scss file cleaner.  100% my fault for forgetting about it.
